core/src/apps/cardano/sign_tx.py

```diff
@@ -524,10 +534,20 @@ async def _show_outputs(
     total_amount = 0
     for output in msg.outputs:
         if output.address_parameters:
-            if not SCHEMA_ADDRESS.match(output.address_parameters.address_n):
+            if fail_if_true_and_strict(
+                not SCHEMA_ADDRESS.match(output.address_parameters.address_n),
+                "Invalid change output path",
+            ):
                 await show_warning_change_output_path(
                     ctx, output.address_parameters.address_n
                 )
+            if not SCHEMA_ADDRESS.match(output.address_parameters.address_n):
+                await fail_if_strict(
```

Comment:
IMHO passing the function to fail_if_strict() is kind of an overkill. I would just pass the error message and check for the strict flag and subsequently show the prompt in a separate call - IMHO the resulting code would be simpler to follow and it would also be more flexible as the pattern would cover also functions that take parameters without sacrificing type-check which have a hard time with function references

Reply:
in theory, fail_if_strict can be a decorator, but imho it's not worth it as it would ultimately conceal an arguably relevant piece of logic